<?xml version="1.0"?>
<robot name="PuzzleBot">
  <!-- Base link -->
<link name="base_link">
  <visual>
    <origin xyz="0 0 0" />  <!-- Desplazamiento de 0.05 unidades en el eje Y -->
    <geometry>
      <box size="0 0 0"/>  <!-- Nota: esta caja no tiene tamaño definido, considera especificar las dimensiones -->
    </geometry>
  </visual>
</link>


<!-- Chassis link -->
<link name="chassis">
  <visual>
    <origin rpy="1.5708 0 1.5708" xyz="0 0 0.05" /> <!-- Rotación de 90 grados y desplazamiento de 5 cm hacia arriba -->
    <geometry>
      <mesh filename="package://localisation/models/MCR2_1000_13_Chassis.stl"/>
    </geometry>
  </visual>
  <collision>
    <origin rpy="1.5708 0 1.5708" xyz="0 0 0.05" /> <!-- Rotación de 90 grados y desplazamiento de 5 cm hacia arriba -->
    <geometry>
      <mesh filename="package://localisation/models/MCR2_1000_13_Chassis.stl"/>
    </geometry>
  </collision>
</link>



<!-- Wheel Coupler link -->
<link name="wheel_coupler">
  <visual>
    <origin rpy="0 1.5708 -1.5708" xyz="0.05 0.085 0.05" /> <!-- Rotación de 90 grados y desplazamiento de 5 cm hacia arriba -->
    <geometry>
      <mesh filename="package://localisation/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
    </geometry>
  </visual>
</link>

<!-- Wheel Coupler 2 link -->
<link name="wheel_coupler_2">
  <visual>
    <origin rpy="0 -1.5708 -1.5708" xyz="0.05 -0.085 0.05" /> <!-- Rotación de 90 grados y desplazamiento de 5 cm hacia arriba -->
    <geometry>
      <mesh filename="package://localisation/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
    </geometry>
  </visual>
</link>



  <!-- Joints -->
  <!-- Static joint connecting base_link to chassis -->
  <joint name="chassis_joint" type="fixed">
    <parent link="base_link"/>
    <child link="chassis"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>

  <!-- Assuming wheel couplers are where the wheels would be connected -->
  <joint name="wheel_coupler_joint" type="fixed">
    <parent link="chassis"/>
    <child link="wheel_coupler"/>
    <!-- You will need to adjust the origin based on the actual position -->
    <origin xyz="0 0 0" rpy="0 0 0"/>
  
  </joint>
    <joint name="wheel_coupler_joint_2" type="fixed">
    <parent link="chassis"/>
    <child link="wheel_coupler_2"/>
    <!-- You will need to adjust the origin based on the actual position -->
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>
</robot>